### PR TITLE
Moving CSV::Row to their own file

### PR DESCRIPTION
**Pull request by esparta**
Implements part of https://github.com/ruby/csv/issues/5

This does not change any behavior of the csv gem.

### CODE REVIEW

File: lib/csv.rb [hunk 1]
```diff
@@ -589,6 +215,8 @@ def inspect
   # All tables returned by CSV will be constructed from this class, if header
   # row processing is activated.
   #
+
+  require_relative "csv/row"
```
Comment (kou) [inline lib/csv.rb]: We should put `require` family at the top level not in `class`.

File: lib/csv/row.rb [hunk 2]
```diff
@@ -0,0 +1,384 @@
+# frozen_string_literal: true
+# = csv.rb -- CSV Reading and Writing
```
Comment (kou) [inline lib/csv/row.rb]: This file isn't `csv.rb`.
We can remove the header description by moving the description to README.

**Comment by esparta** [comment]
Done with the changes requested. Thanks, @kou !

File: lib/csv.rb [hunk 3]
```diff
@@ -589,6 +216,7 @@ def inspect
   # All tables returned by CSV will be constructed from this class, if header
   # row processing is activated.
   #
+
```
Comment (stevendaniels) [inline lib/csv.rb]: Should we remove this extra line-break?

Comment (kou) [reply lib/csv.rb]: @esparta Can you remove this change?

Comment (esparta) [reply lib/csv.rb]: sure, no problem

**Comment by kou** [comment]
Thanks!

**Pull request merged**